### ISSUE TITLE
Insert a value when deleteing automation on row 0

### ISSUE THREAD

**Issue by offten** [JoepVanlier/Hackey-Trackey#99]
As it is right now, a big issue for me when using the app for automation is if you accidentally hit '.' on row zero. It enters a 00 but Reaper allows automation at the beginning of a section to be absent and simply jumps to the next point. So while I remember you saying that this is a limitation of Reaper, I think the best thing to do would be to force a point to be created the first row. Right now you have to go into the automation in Reaper and create an initial point, then re-launch hacky-tracky to fix it.

**Comment by JoepVanlier** [comment]
I'm not sure what you mean. If you hit `..` at row zero, it just deletes that point. So if reaper was set to interpolate from the last one, it just continues as far as I know?

I'm not sure why you need to go into the automation in REAPER to create a new initial point. Why can't you just enter it from within HT? I'm not opposed to adding a flag in the config for auto-adding a first point, but what value would that point need to have?

I think I don't understand exactly what you mean, and what your intended workflow is. Could you maybe make a short screen recording?

**Comment by offten** [comment]
Yeah sorry I was gonna delete this because I think what I'm doing wrong is creating a media item at the _beginning_ of a project. Otherwise the first point can be deleted without issues. Might also be something in my automation settings or in your settings that pushes media items forwards / backwards. I only suspect this because sometimes Reaper will fail to trigger the first hit of midi in the piano roll. So it might be a Reaper issue and the workaround is to move items forward a bit which is fine.

**Comment by JoepVanlier** [comment]
Ah, I have just reproduced that weirdness at time point zero. That is indeed odd. It seems that the automation system returns `-1` for the "pre" envelope point. I have just pushed a partial workaround that hopefully is workable for you in `3.27`. Basically, it should at least allow you to overwrite that value now.
